flaskr_tests.py

```diff
@@ -1,48 +1,66 @@
+"""Test application.
+
+There are tests intended to check that app is functioning correctly
+"""
+
 import os
 import flaskr
 import unittest
 import tempfile
+from bs4 import BeautifulSoup
+
 
 class FlaskrTestCase(unittest.TestCase):
+    """Test class."""
 
     def setUp(self):
+        """Set the app up."""
         self.db_fd, flaskr.app.config['DATABASE'] = tempfile.mkstemp()
         flaskr.app.config['TESTING'] = True
         self.app = flaskr.app.test_client()
         with flaskr.app.app_context():
             flaskr.init_db()
 
     def tearDown(self):
+        """Close all connections."""
         os.close(self.db_fd)
         os.unlink(flaskr.app.config['DATABASE'])
 
     def test_empty_db(self):
+        """Test with empty database."""
         rv = self.app.get('/')
         assert b'No entries here so far' in rv.data
 
     def login(self, username, password):
-        return self.app.post('/login', data=dict(
-            username=username,
-            password=password
-            ), follow_redirects=True)
+        """Handy method to send the authentication data."""
+        return self.app.post(
+            '/login',
+            data=dict(
+                username=username,
+                password=password),
+            follow_redirects=True)
 
     def logout(self):
+        """Log the admin user out."""
         return self.app.get('logout', follow_redirects=True)
 
     def test_login_logout(self):
-        rv = self.login('admin','default')
+        """Login/logout test."""
```

Comment:
It should be something like "It is possible to login and logout", see this post on how to write the docsting for tests -https://jml.io/pages/test-docstrings.html.

Also it is better to have each case tested in the separate (small) test: test_login, test_logout, test_invalid_username, test_invalid_password.